### PR TITLE
EXLM-96 AEM UE Integration with converter

### PR DESCRIPTION
**Pull request by ruhisingh1**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -73,6 +75,46 @@ const renderFragment = async (path) => {
   };
 };
 
+/**
+ * Renders pages from AEM
+ */
+const renderAEMPages = async (path, params) => {
+  const { authorization, wcmmode } = params;
+  const aemURL = `${aemConfig.aemEnv}${path}.html`;
```
Comment (Sivaramvt) [inline src/index.js:83]: Append wcmmode=disabled to the AEM URL to get rid of cq component decoration tags

Comment (ruhisingh1) [reply src/index.js]: Fixed

File: src/index.js [hunk 2]
```diff
@@ -81,6 +126,10 @@ export const render = async function render(path) {
   if (path.startsWith('/fragments')) {
     return renderFragment(path);
   }
+  // Handle AEM UE Pages
+  if (path.startsWith('/content')) {
```
Comment (Buuhuu) [inline src/index.js:130]: You probably don't want to include `/content/...` in the urls right?

I would suggest to request everything that is not /docs or /fragments from AEM.

File: src/index.js [hunk 3]
```diff
@@ -73,6 +75,49 @@ const renderFragment = async (path) => {
...
+ */
+const renderContent = async (path, params) => {
+  const { authorization } = params;
+
+  /**
+   * FIXME: Page extension to be handled dynamically. Revisit this implementation and identify a proper
+   * fix.
+   * */
+
+  const aemURL = `${aemConfig.aemEnv}${path}.html?wcmmode=disabled`;
```
Comment (Buuhuu) [inline src/index.js:89]: You should call the franklin delivery servlet which does some important post processing on the html

```
`${aemConfig.aemEnv}/bin/franklin.delivery/${owner}/${repo}/${ref}/${path}?wcmmode=disabled`;
```

AEM will take care of the reverse mapping of the `path`.

**Review by Buuhuu — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 4]
```diff
@@ -73,6 +75,49 @@ const renderFragment = async (path) => {
...
+  }
+
+  let contentType = resp.headers.get('content-type') || 'text/html';
+  [contentType] = contentType.split(';');
+
+  const respHeaders = {
+    'content-type': contentType,
+  };
+
+  if (isBinary(contentType)) {
```
Comment (Buuhuu) [inline src/index.js:111]: We should check if this is really necessary. Ideally we would not spool binaries through the converter. That means we rather check if the incoming request is .html or .json and if not fail with a [415 Unsupported Media Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/415)

File: src/index.js (hunk 2 above)
Comment (ruhisingh1) [reply src/index.js]: Addressed

File: src/index.js (hunk 3 above)
Comment (ruhisingh1) [reply src/index.js]: Addressed

**Review by Sivaramvt — APPROVED** [review]

**Pull request merged**